src/data/use-cases/schedule/tag/db-update-tag.ts

```diff
@@ -0,0 +1,27 @@
+import { NotAcceptableError } from '@/application/errors'
+import { type LoadTagByIdRepository, type UpdateTagRepository } from '@/data/protocols'
+import { type UpdateTag } from '@/domain/use-cases/schedule/tag'
+
+export class DbUpdateTag implements UpdateTag {
+  private readonly tagRepository: UpdateTagRepository & LoadTagByIdRepository
+
+  constructor ({ tagRepository }: UpdateTag.Dependencies) {
+    this.tagRepository = tagRepository
+  }
+
+  async update (params: UpdateTag.Params): Promise<UpdateTag.Result> {
+    const { id, name } = params
```

Comment:
Pode fazer a desestruturação antes de entrar no metodo
```suggestion
  async update ({ id, name }: UpdateTag.Params): Promise<UpdateTag.Result> {
```